pkg/process/scheduler/exclusive.go

```diff
@@ -0,0 +1,107 @@
+package scheduler
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+var _ Scheduler = &exclusiveScheduler{}
+
+func NewExclusiveScheduler() Scheduler {
+	return &exclusiveScheduler{}
+}
+
+// exclusiveScheduler is a scheduler that runs a single process at a time.
+// Does not support concurrent execution of multiple processes.
+type exclusiveScheduler struct {
```

Comment:
Scheduler is big concept, we should simplify the concept as much as possible. Here it's better to name it ExclusiveProcess, and if it's running, we can return an error. In other words, this control logic is the responsibility of the caller.